examples/cfs-throttling.bpf.c

```diff
@@ -0,0 +1,34 @@
+#include <vmlinux.h>
+#include <bpf/bpf_tracing.h>
+#include "bits.bpf.h"
+#include "maps.bpf.h"
+
+// 21 buckets for latency, max range is 0.5s .. 1.0s
+#define MAX_LATENCY_SLOT 20
+
+#define MAX_CGROUPS 1024
+
+struct key_t {
+    u32 cgroup;
+    u32 bucket;
+};
+
+struct {
+    __uint(type, BPF_MAP_TYPE_LRU_HASH);
+    __uint(max_entries, (MAX_LATENCY_SLOT + 1) * MAX_CGROUPS);
+    __type(key, struct key_t);
+    __type(value, u64);
+} cfs_throttling_seconds SEC(".maps");
+
+SEC("fentry/unthrottle_cfs_rq")
+int BPF_PROG(unthrottle_cfs_rq, struct cfs_rq *cfs_rq)
+{
+    u64 throttled_us = (cfs_rq->rq->clock - cfs_rq->throttled_clock) / 1000;
```

Comment:
`cfs_rq->throttled_clock` can be zero, at least in newer kernels:

* https://github.com/torvalds/linux/commit/79462e8c879afc7895b30014d31e2c1fd629bb1f